pkgs/libresoc/nmigen-soc.nix

```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  python39Packages,
+  fetchFromGitLab,
+  nmigen,
+}:
+with python39Packages;
+  buildPythonPackage rec {
+    pname = "nmigen-soc";
+    version = "unstable-2024-03-31";
+    # python setup.py --version
+    realVersion = "0.1.dev243+g${lib.substring 0 7 src.rev}";
+
+    # NOTE(jleightcap): while libre-soc project does have local forks of nmigen* projects,
+    # HEADs of repos are incompatible.
+    # dev-env-setup implies that these forks are unused in build process, so using upstream.
+    src = fetchFromGitLab {
+      owner = "nmigen";
+      repo = "nmigen-soc";
```

Comment:
nmigen-soc was not currently packaged in nixpkgs during work on this PR. However, it was also renamed to https://github.com/amaranth-lang/amaranth-soc and has now been packaged in nixpkgs, but suffers from the same rename issue as the previous commit with `nmigen` and `amaranth`